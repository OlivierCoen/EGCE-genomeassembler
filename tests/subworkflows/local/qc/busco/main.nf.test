nextflow_workflow {

    name "Test Workflow QC_BUSCO"
    script "subworkflows/local/qc/busco/main.nf"
    workflow "QC_BUSCO"
    tag "busco_subworkflow"

    test("Should run without failures") {

        when {

            params {
                busco_lineage = "arthropoda_odb10"
            }

            workflow {
                """
                input[0] = [
                    [ id: "test", genome_size: 1500000 ],
                    file( '$projectDir/.nf-test/tests/d1e58da82b7582837d1ff22e9f90967d/work/dc/d26d605e9f5a209b168ff0902be97f/results/6-polish/medaka/alternate.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
