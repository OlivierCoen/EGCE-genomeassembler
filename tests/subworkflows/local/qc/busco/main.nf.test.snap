{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "genome_size": 1500000
                        },
                        "test-arthropoda_odb10-busco.batch_summary.txt:md5,607a99c2a2569def751a155be09d4876"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "genome_size": 1500000
                        },
                        "short_summary.specific.arthropoda_odb10.alternate.fasta.json:md5,49c05c15700f1abacfb5da2db5a1e3a1"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "genome_size": 1500000
                        },
                        "short_summary.specific.arthropoda_odb10.alternate.fasta.txt:md5,bb3c59bbe5cba1428752b8ce754db581"
                    ]
                ],
                "3": [
                    "versions.yml:md5,d4bac71d61e8e2f360fba3d461b004c7"
                ],
                "batch_summary": [
                    [
                        {
                            "id": "test",
                            "genome_size": 1500000
                        },
                        "test-arthropoda_odb10-busco.batch_summary.txt:md5,607a99c2a2569def751a155be09d4876"
                    ]
                ],
                "short_summary_json": [
                    [
                        {
                            "id": "test",
                            "genome_size": 1500000
                        },
                        "short_summary.specific.arthropoda_odb10.alternate.fasta.json:md5,49c05c15700f1abacfb5da2db5a1e3a1"
                    ]
                ],
                "short_summary_txt": [
                    [
                        {
                            "id": "test",
                            "genome_size": 1500000
                        },
                        "short_summary.specific.arthropoda_odb10.alternate.fasta.txt:md5,bb3c59bbe5cba1428752b8ce754db581"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d4bac71d61e8e2f360fba3d461b004c7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-22T17:27:36.912786296"
    }
}